### DISCUSSION TITLE
Is it possible to add a `breakable` for cross referenced Code Block?

### DISCUSSION THREAD

**Discussion by joelvonrotz** [quarto-dev/quarto-cli#8527] [Q&A] [answered]
### Description

Hello everyone

I'm in a bit of a pickle. After resolving #8519 , I'm trying to solve the next *problem*. The goal is to have a cross referenced, syntax highlighted code block which is multiple pages long. Right now, to have the syntax highlighting AND be cross referenced, the code block is put inside a [cross referenced div](https://quarto.org/docs/authoring/cross-references-divs.html). Now is it possible to adjust the `codelisting` float environment to correctly break up the code block?

````qmd
---
title: syntax
format:
  pdf:
    pdf-engine: pdflatex
    include-in-header: 
    - text: "\\usepackage{blindtext}"
---

:::{#lst-hello2}


```cpp
int8_t hello;

int8_t add(int8_t a, int8_t b);
int8_t subtract(int8_t a, int8_t b);

void main(void) {
  hello = 1;

  while(true) {
    hello++;
    if(hello=10) {
      hello = 0;
    }
  }
}

int8_t add(int8_t a, int8_t b) {
  return a + b;
}

int8_t subtract(int8_t a, int8_t b) {
  return a - b;
}
int8_t hello;

int8_t add(int8_t a, int8_t b);
int8_t subtract(int8_t a, int8_t b);

void main(void) {
  hello = 1;

  while(true) {
    hello++;
    if(hello=10) {
      hello = 0;
    }
  }
}

int8_t add(int8_t a, int8_t b) {
  return a + b;
}

int8_t subtract(int8_t a, int8_t b) {
  return a - b;
}
int8_t hello;

int8_t add(int8_t a, int8_t b);
int8_t subtract(int8_t a, int8_t b);

void main(void) {
  hello = 1;

  while(true) {
    hello++;
    if(hello=10) {
      hello = 0;
    }
  }
}

int8_t add(int8_t a, int8_t b) {
  return a + b;
}

int8_t subtract(int8_t a, int8_t b) {
  return a - b;
}
int8_t hello;

int8_t add(int8_t a, int8_t b);
int8_t subtract(int8_t a, int8_t b);

void main(void) {
  hello = 1;

  while(true) {
    hello++;
    if(hello=10) {
      hello = 0;
    }
  }
}

int8_t add(int8_t a, int8_t b) {
  return a + b;
}

int8_t subtract(int8_t a, int8_t b) {
  return a - b;
}
```

Hello #2

:::



```{=latex}
\blindtext

\blindtext
```

@lst-hello2
````

## Example of the end use

![image](https://github.com/quarto-dev/quarto-cli/assets/41155337/8905c4fd-b672-4284-ba8e-e30000c78c25)

What I like to have (with cross referencing)

![image](https://github.com/quarto-dev/quarto-cli/assets/41155337/6f4ca27a-8a5e-44da-b717-00817f942f1b)

I know this behavior is due to the usage of a float environment, but is there a possible workaround?

**Comment by mcanouil** [comment]
That's a LaTeX question.
You'll have to tinker LaTeX code directly. Use `keep-tex: true` and play with the LaTeX file.

I am not even sure it's possible in plain LaTeX with the current output.

**Comment by cscheid** [comment, marked as answer]
As far as I can tell, it's not. page-spanning environments in LaTeX are complicated (there's `longtable` for tables and that by itself introduces a ton of extra complications in the typesetting)

**Comment by joelvonrotz** [comment]
Ah okay, that makes sense. I've found a workaround for now. By creating an empty cross reference div, remove a bit of vertical spacing between the caption and the code block and add the caption into the div-information, I kind of get what I wanted!

````qmd

:::{#lst-hello2 lst-cap="hello"}
:::

`\vspace{-4mm}`{=latex}

```cpp
...

```
````

![image](https://github.com/quarto-dev/quarto-cli/assets/41155337/141ced1e-3011-447c-9d8c-f15b92117c3d)
